api/src/main/kotlin/io/spine/protodata/renderer/SourceFileSet.kt

```diff
@@ -316,3 +328,62 @@ private fun checkTarget(targetRoot: Path) {
         }
     }
 }
+
+/**
+ * Part of the fluent API for finding source files.
+ */
+public class FileLookup(
+    private val sources: SourceFileSet,
+    private val name: TypeName
+) {
+
+    /**
+     * Searches for a source file with for the given Proto type generated according to
+     * the given [convention].
+     */
+    public fun <L: Language, N: TypeNameElement<L>> namedUsing(
+        convention: TypeConvention<L, N>
+    ): SourceFile? {
+        val declaration = convention.declarationFor(name)
+        val path = declaration?.path
+        return path?.let { sources.find(it) }
+    }
+}
+
+/**
+ * Part of the fluent API for creating new source files.
+ */
+public class FileCreation(
+    private val sources: SourceFileSet,
+    private val name: TypeName
+) {
+
+    /**
+     * Attempts to create a file path for the given type name using the given [convention].
+     *
+     * If the convention does not define a declaration for the given type, returns `null`.
+     */
+    public fun <L: Language, N: TypeNameElement<L>> namedUsing(
+        convention: TypeConvention<L, N>
+    ): FileCreationWithPath? {
+        val declaration = convention.declarationFor(name)
+        val path = declaration?.path
+        return path?.let { FileCreationWithPath(sources, path) }
+    }
+}
+
+/**
+ * Part of the fluent API for creating new source files.
+ */
+public class FileCreationWithPath(
```

Comment:
I think `FileLookup`, `FileCreation`, and `FileCreationWithPath` deserve a common interface which would group them semantically. These classes look like operations on `SourceFileSet`, and they all have `sources` as a property. 

I understand that they do not have a common API in terms of producing the output. And that could be the reason of not having a common interface for them. Still, grouping interface would help in documenting and discovering those operations.